### PR TITLE
Add rh-ruby binaries to path on centos-7-rpmpkg

### PR DESCRIPTION
**Pull request by adiaaida**
For the rpmpkg container, we need to be able to find fpm, which is now being installed within scl, and no longer in the standard place. This change adds the rh-ruby bin paths to the PATH, LD_LIBRARY_PATH, and a few others so that runtime can generate rpms on in this container.

### CODE REVIEW

File: src/centos/7/rpmpkg/Dockerfile [hunk 1]
```diff
@@ -1,12 +1,20 @@
...
+    && yum --enablerepo=centos-sclo-rh -y install ${RUBY_SCL}-ruby-devel \
+    && yum --enablerepo=centos-sclo-rh -y install ${RUBY_SCL}-rubygems \
     && yum install -y \
         gcc \
         rpm-build \
-    && scl enable rh-ruby26 'gem install --no-document fpm git:1.7.0 ffi:1.12.2' \
+    && scl enable ${RUBY_SCL} 'gem install --no-document fpm git:1.7.0 ffi:1.12.2' \
     && yum clean all
+
+ENV PATH="/opt/rh/${RUBY_SCL}/root/usr/local/bin:/opt/rh/${RUBY_SCL}/root/usr/bin:${PATH}" \
```
[the remark this answers is not in the document]
Comment (adiaaida) [reply src/centos/7/rpmpkg/Dockerfile]: Fixed. I wasn't sure if there was a way to symlink the libraries, so I left in the LD_LIBRARY_PATH (fpm fails without it). Everything else was unnecessary clutter to fix runtime

File: src/centos/7/rpmpkg/Dockerfile [hunk 2]
```diff
@@ -1,12 +1,20 @@
...
+    LD_LIBRARY_PATH="/opt/rh/${RUBY_SCL}/root/usr/local/lib64:/opt/rh/${RUBY_SCL}/root/usr/lib64" \
+    X_SCLS="scl enable ${RUBY_SCL}" \
+    XDG_DATA_DIRS="/opt/rh/${RUBY_SCL}/root/usr/local/share:/opt/rh/${RUBY_SCL}/root/usr/share:/usr/local/share:/usr/share" \
+    PKG_CONFIG_PATH="/opt/rh/${RUBY_SCL}/root/usr/local/lib64/pkgconfig:/opt/rh/${RUBY_SCL}/root/usr/lib64/pkgconfig"
```
[the remark this answers is not in the document]
Comment (adiaaida) [reply src/centos/7/rpmpkg/Dockerfile]: Done

File: src/centos/7/rpmpkg/Dockerfile [hunk 3]
```diff
@@ -1,12 +1,21 @@
...
-    && scl enable rh-ruby26 'gem install --no-document fpm git:1.7.0 ffi:1.12.2' \
+    && scl enable ${RUBY_SCL} 'gem install --no-document fpm git:1.7.0 ffi:1.12.2' \
     && yum clean all
+
+# Create symlinks for fpm and ruby so they are on the path
+RUN ln -sf /opt/rh/${RUBY_SCL}/root/usr/local/bin/fpm /usr/bin/fpm \
+    && ln -sf /opt/rh/${RUBY_SCL}/root/usr/bin/ruby /usr/bin/ruby
+
+# Set up additional environment variables so fpm can run
+ENV LD_LIBRARY_PATH="/opt/rh/${RUBY_SCL}/root/usr/lib64:/opt/rh/${RUBY_SCL}/root/usr/local/lib64" \
```
Comment (mthalman) [inline src/centos/7/rpmpkg/Dockerfile:21]:
```suggestion
ENV LD_LIBRARY_PATH="/opt/rh/${RUBY_SCL}/root/usr/lib64:/opt/rh/${RUBY_SCL}/root/usr/local/lib64"
```

**Comment by wfurt** [comment]
cc: @omajid in case there is more standard way to do it.

**Comment by aslicerh** [comment]
Hey all, in the .NET images, we set some environment variables that run a script that does `scl enable` automatically, for an interactive or non-interactive prompt.

For an example, see [here](https://github.com/redhat-developer/s2i-dotnetcore/blob/master/3.1/runtime/Dockerfile#L59-L68) in our Centos dockerfile, and [here](https://github.com/redhat-developer/s2i-dotnetcore/blob/master/3.1/runtime/contrib/etc/scl_enable) for the script.

It's also nice because it is expandable by a user, as they can build on top of our image and add to the `ENABLED_COLLECTIONS` env var if they need to add their own scl collection on top of what we provide out of the box (for example, NodeJS).

**Comment by adiaaida** [comment]
> Hey all, in the .NET images, we set some environment variables that run a script that does `scl enable` automatically, for an interactive or non-interactive prompt.
> 
> For an example, see [here](https://github.com/redhat-developer/s2i-dotnetcore/blob/master/3.1/runtime/Dockerfile#L59-L68) in our Centos dockerfile, and [here](https://github.com/redhat-developer/s2i-dotnetcore/blob/master/3.1/runtime/contrib/etc/scl_enable) for the script.
> 
> It's also nice because it is expandable by a user, as they can build on top of our image and add to the `ENABLED_COLLECTIONS` env var if they need to add their own scl collection on top of what we provide out of the box (for example, NodeJS).

Thanks for the feedback @aslicerh. That's super helpful, and seems to work for our scenario as well. The only weird thing that I can't figure out (given that I almost directly copied the example) is that I'm getting this text when I start up the container:

```
': not a valid identifier
```

I'm not sure where it is coming from. I will push my current branch and maybe someone will see something I'm missing?

**Comment by aslicerh** [comment]
@adiaaida That's odd. I'm not sure where that would be coming from.

I tried to replicate it myself, though I'm having to change some things in the Dockerfile because the base image you're using doesn't resolve for me. [Here's my Dockerfile](https://github.com/aslicerh/dotnet-buildtools-prereqs-docker/blob/centos_ruby_dockerfile_error/src/centos/7/rpmpkg/Dockerfile). I can't actually recreate it. When I build from mine, I don't get any error like that.

**Comment by adiaaida** [comment]
Thanks for trying to repro the issue. Maybe it's something to do with one of the base images. I'll check.

**Comment by adiaaida** [comment]
It appears that my version of Docker (docker on windows, if that matters) does not like the unset command in scl_enable.

**Comment by aslicerh** [comment]
@adiaaida Ah, is that using WSL? I'm using native linux here and don't really have a way to test stuff on Windows. Still, if you're using bash for WSL I would expect it to work the same. Or are you using something like plain sh, zsh, or something like that?

**Comment by adiaaida** [comment]
The base container is the centos-7 container created by https://github.com/dotnet/dotnet-buildtools-prereqs-docker/blob/main/src/centos/7/Dockerfile. unset works fine within the container if I just run it, but for whatever reason, it being in the script run by PROMPT_COMMAND causes the weird message that I am seeing. fpm (the dependency that we're trying to get on the path) will run despite the output message.

As for how I'm running the containers, it's the Docker Desktop app.

**Comment by adiaaida** [comment]
The issue was that the file saved on my box was using windows line endings. I fixed that, and it's happy now. This is ready for final review

**Review by mthalman — APPROVED** [review]

**Pull request merged**